### ISSUE TITLE
effectless endpoint kills zio fiber

### ISSUE THREAD

**Issue by Phill101** [finagle/finch#1138]
Hi. Trying to use finch with zio.Task (through cats-interop). But I've met a problem. 
```scala
get("div" :: path[Int] :: path[Int]) { (a: Int, b: Int) =>
  Ok((a / b).toString)
} handle {
  case e: Exception => BadRequest(e)
}
```
This simple endpoint kills fiber, when we try divide by zero. `handle` block do nothing and client receive 500.
If we wrap expression in Task.effect constructor all works as expected.

Seems like it happens in Output.traverseFlatten:55 when Endpoint.mapOutputAsync:109.
In cats.IO, and monix.Task you can throw an exception in functions that you pass as an argument. But not in ZIO. There is `map` awaits for a pure function. And it seems reasonable for me.

Can we refactor this two methods? I would like to try to help

**Comment by Phill101** [comment]
#1142

**Comment by Phill101** [comment]
Added test. Probably it has not better name.

**Comment by sergeykolbasov** [comment]
Fixed in #1142
